src/mbgl/map/map_context.cpp

```diff
@@ -299,14 +310,16 @@ void MapContext::onLowMemory() {
 }
 
 void MapContext::setSprite(const std::string& name, std::shared_ptr<const SpriteImage> sprite) {
+    assert(util::ThreadContext::currentlyOn(util::ThreadType::Map));
     if (!style) {
```

Comment:
It's no longer necessary to have a style before adding an annotation sprite -- we can now remove this block!